src/dds/rs-dds-device-proxy.cpp

```diff
@@ -145,7 +146,7 @@ dds_device_proxy::dds_device_proxy( std::shared_ptr< context > ctx, std::shared_
             if( ! sensor_info.proxy )
             {
                 // This is a new sensor we haven't seen yet
-                sensor_info.proxy = create_sensor( stream->sensor_name() );
+                sensor_info.proxy = create_sensor( stream->sensor_name(), stream->type_string() );
```

Comment:
What will happen on sensors that are have color and depth streams (D405,D505)? The first stream will set the entire sensor type?

Reply:
Good catch!

I've redone it as discussed: we go over all the streams and decide the format for the sensor, then create the sensor class based on the format (and not the name).

The mock D405 still needs work -- I didn't do it as part of this PR. I already have many commits that I decided not to push in here, so maybe in the next PR...